cvmfs/cvmfs_server

```diff
@@ -825,14 +832,25 @@ is_owner_or_root() {
 # a specific CVMFS repository.
 #
 # @param name   the name of the repository whose owner should be impersonated
-# @return       a shell invocation to impersonate $CVMFS_USER
+# @return       a shell invocation to impersonate $CVMFS_USER via stdout or
+#               exit code 1 if user couldn't be impersonated
 get_user_shell() {
   local name=$1
+  local shell_cmd=""
 
   load_repo_config $name
-  local user_shell="sh -c"
-  [ $(whoami) != $CVMFS_USER ] && user_shell="sudo -E -H -u $CVMFS_USER sh -c"
-  echo "$user_shell"
+  if [ x"$(whoami)" = x"$CVMFS_USER" ]; then
+    shell_cmd="sh -c"
+  elif is_root; then
+    if [ $HAS_RUNUSER -ne 0 ]; then
+      shell_cmd="runuser -m $CVMFS_USER -c"
+    else
+      shell_cmd="su $CVMFS_USER -c"
```

Comment:
Apparently not. Citing the [man page](http://man7.org/linux/man-pages/man1/su.1.html):
```
       For backward compatibility, su defaults to not change the current
       directory and to only set the environment variables HOME and SHELL
       (plus USER and LOGNAME if the target user is not root).  It is
       recommended to always use the --login option (instead of its shortcut
       -) to avoid side effects caused by mixing environments.
```